reporter/src/main/kotlin/reporters/AsciidocAttributionDocumentReporter.kt

```diff
@@ -0,0 +1,102 @@
+ * For each Freemarker template provided using the options described below a separate intermediate file is created
+ * that can be processed by AsciidoctorJ. If no options are provided the default "asciidoc" template is used.
+ * The name of the template id or template path (without extension) is used for the intermediate file,
+ * so be careful to not use two different templates with the same name.
+ *
+ * After the intermediate files are generated, they are processed by AsciidoctorJ or to be more precise by its pdf
+ * implementation AsciidoctorJ PDF. A pdf theme can be handed over to AsciidoctorJ PDF in which properties like fonts
+ * or images displayed in the pdf can be adjusted; see the [Theme Guide][5].
+ * The path to this theme can be set in the options as described below.
+ * Note that only one theme can be set that is used for all given templates. If no theme is given or the given path to
+ * the theme file does not exist, an in-built theme of AsciidoctorJ PDF is used.
+ *
+ * This reporter supports the following options:
+ * - *template.id*: A comma-separated list of IDs of templates provided by ORT. Currently only the "asciidoc" template
+ *                  is available.
+ * - *template.path*: A comma-separated list of paths to template files provided by the user.
+ * - *pdf-theme.path*: A path to an Asciidoc pdf theme file.
+ *
+ * [1]: https://freemarker.apache.org
+ * [2]: https://asciidoc.org/
+ * [3]: https://github.com/asciidoctor/asciidoctorj
+ * [4]: https://github.com/asciidoctor/asciidoctorj-pdf
+ * [5]: https://github.com/asciidoctor/asciidoctor-pdf/blob/master/docs/theming-guide.adoc
+ */
+class AsciidocAttributionDocumentReporter : Reporter {
+    companion object {
+        private const val ASCIIDOCS_FILE_PREFIX = "Asciidocs_"
+        private const val ASCIIDOCS_DEFAULT_TEMPLATE = "asciidoc"
+
+        private const val OPTION_PDF_THEME_PATH = "pdf-theme.path"
+    }
+
+    val templateGenerator = FreemarkerTemplateProcessor(ASCIIDOCS_FILE_PREFIX, ASCIIDOCS_DEFAULT_TEMPLATE)
+
+    override val reporterName = "Asciidoc"
+
+    override fun generateReport(input: ReporterInput, outputDir: File, options: Map<String, String>): List<File> {
+        val asciidocsInput = templateGenerator.generateOutput(input, outputDir, options)
+
+        val asciidoctor = create()
+        val outputFiles = mutableListOf<File>()
+
+        val asciidoctorAttributes = AttributesBuilder.attributes()
+        val themePath = options[OPTION_PDF_THEME_PATH]
+
+        if (themePath != null) {
+            File(themePath).also {
+                require(it.exists()) { "Could not find theme file at ${it.absolutePath}." }
+            }
+            asciidoctorAttributes.attribute("pdf-theme", themePath)
+        }
+
+        asciidocsInput.forEach { file ->
+            val outputFile = outputDir.resolve("${file.nameWithoutExtension}.pdf")
+            outputFiles += outputFile
```

Comment:
Nit: Let's make this line the last operation in the loop, so the file gets only added after it was successfully created.